### DISCUSSION TITLE
Automatic version update

### DISCUSSION THREAD

**Discussion by mateus4k** [scripthookvdotnet/scripthookvdotnet#1420] [Ideas]
When a new GTA version is released, what you need to do is to update the version enums, compile, and try to run scriphookv? If I'm right, I would like to suggest a github actions job which run some days of week and find for a new GTA version, then open a pull request with enum changes and generate an alpha build version to users. What do you think about it?

**Comment by Michael21107** [comment]
script hook v? Do you realize this is scripthookv**dotnet**, a different script?

**Comment by kagikn** [comment]
We might as well manual update `GameVersion` enums, for at least 2 reasons:
* There's no **RELIABLE** ways for shell scripts to fetch the latest game version in the format like v1.0.2699.0, not the online version such as 1.56. Fetching the version info from the exe is not acceptable, because we can't make sure it's the latest one. There's those who prefer playing some old version and there's a way, at least for v1.0.2060.0 or later and v1.0.372.2.
* Each *defined* enum members of `GameVersion` enums should match what `getGameVersion` of the SHV API returns for specific game versions

We *have* to wait for SHV getting updated before we can make sure what `getGameVersion` of SHV returns for new game version(s). I would say manual updating the enums would be a 5-minute job. Also, `Game.Version` that SHVDN returns the same as what SHV's `getGameVersion` returns but with human-readable names. Even if the enums don't define one you want to use, you can still use by casting to a int, like;
```cs
bool IsGameVersion2699OrEarlier() => Game.Version <= (GameVersion)78; // 78 is what `getGameVersion` returns if you are using v1.0.2699.0 non-Steam
```
